lib/mockito.dart

```diff
@@ -48,6 +53,119 @@ class Mock {
   String toString() => _givenName != null ? _givenName : runtimeType.toString();
 }
 
+// Return a new [Invocation], reconstituted from [invocation], [_typedArgs],
+// and [_typedNamedArgs].
+Invocation _reconstituteInvocation(Invocation invocation) {
+  var newInvocation = new FakeInvocation(invocation);
+  _typedArgs.clear();
+  _typedNamedArgs.clear();
+  return newInvocation;
+}
+
+/// An Invocation class with a functional constructor.
+class FakeInvocation extends Invocation {
+  final Symbol memberName;
```

Comment:
Can you cite something regarding this? I wasn't sure either. The analyzer doesn't complain. I thought this was because the field adds an invisible, implicit getter. Do I need the `@override` annotations on such fields?